lib/utils.dart

```diff
@@ -67,21 +68,20 @@ generateDefaultInputValues(featureConfig) {
   return defaultInputs;
 }
 
+/// Creates either a radio button or a slider for [feature].
+/// [context] is the widget that the input widget is on, so the widget that has to reload.
 buildInputWidget(MyHomePageState homePageState, State context,
-    MapEntry<String, dynamic> feature, Map<String, dynamic> userInputs) {
+    MapEntry<String, dynamic> feature) {
   if (feature.value["choices"] != null) {
-    var buttons =
-        getRadioButtonInputRow(homePageState, context, feature, userInputs);
-    return buttons;
+    return getRadioButtonInputRow(homePageState, context, feature);
   } else if (feature.value["slider_min"] != null) {
-    var slider = getSliderInputRow(
-        homePageState, context, feature, userInputs[feature.key]);
-    return slider;
+    return getSliderInputRow(homePageState, context, feature);
   } else {
     throw new Exception("Input Widget not supported: " + feature.key);
   }
 }
 
+/// Returns the color for a [factor] greater 0.
```

Comment:
why is greater 0 important?

Reply:
I think there is no check for that but it will result in an error if you use it with negative numbers.

Reply:
Ah. In my head it sounded like "returns a color, if greater than 0, and something else if not".

```suggestion
/// Returns the color for a [factor]. [factor] should be > 0.
```